docs/components/reports.rst

```diff
@@ -1,4 +1,220 @@
 Reports
 #######
+To add and render custom reports in Mautic, your plugin needs to listen to three events:
 
+- ``\Mautic\ReportBundle\ReportEvents::REPORT_ON_BUILD``
+- ``ReportEvents::REPORT_ON_GENERATE``
+- ``ReportEvents::REPORT_ON_GRAPH_GENERATE``
+
+This guide walks you through defining a custom report, generating report data, and rendering graphs.
+
+Defining the Report
+-------------------
+
+Use the ``ReportEvents::REPORT_ON_BUILD`` event to define:
+
+- The report context
+- Available columns
+- Available filters (defaults to columns)
+- Available graphs
+
+Column Definition
+-----------------
+
+Each column array can include the following properties:
+
+.. vale on
+
+.. list-table::
+    :header-rows: 1
+
+    * - Key
+      - Required?
+      - Type
+      - Description
+    * - ``label``
+      - REQUIRED
+      - string
+      - The language string for the column.
+    * - type
+      - REQUIRED
+      - string
+      - Column type.
+    * - alias
+      - OPTIONAL
+      - string
+      - An alias for the returned value. Useful in conjuction with ``formula``.
+    * - formula
+      - OPTIONAL
+      - string
+      - SQL formula instead of a column. e.g. ``SUBSTRING_INDEX(e.type, \'.\', 1)``.
+    * - link
+      - OPTIONAL
+      - string
+      - Route name to convert the value into a hyperlink. Used usually with an ID of an Entity. The route must accept ``objectAction`` and ``objectId`` parameters.
+
+.. vale off
+
+Filter Definition
+-----------------
+
+Filters are optional. If not defined, Mautic will default to using the column definitions. However, filters can provide additional options such as dropdown select lists.
```

Comment:
This will need reformatting into active voice. The vale off has hidden the error.